Doc/library/threading.rst

```diff
@@ -620,25 +625,38 @@ Reentrant locks also support the :ref:`context management protocol <with-locks>`
 
       Acquire a lock, blocking or non-blocking.
 
-      When invoked without arguments: if this thread already owns the lock, increment
-      the recursion level by one, and return immediately.  Otherwise, if another
-      thread owns the lock, block until the lock is unlocked.  Once the lock is
-      unlocked (not owned by any thread), then grab ownership, set the recursion level
-      to one, and return.  If more than one thread is blocked waiting until the lock
-      is unlocked, only one at a time will be able to grab ownership of the lock.
-      There is no return value in this case.
+      When invoked with the *blocking* argument set to ``True`` (the default):
+
+         1. If another thread owns the lock, block until we are able to acquire
+            lock, or *timeout*, if set.
+
+         2. If no thread owns the lock, acquire the lock and return immediately.
+
+         3. If the same thread owns the lock, acquire the lock again, and
+            return immediately. This is the difference between ``Lock`` and
+            ``RLock``. Internally, the "recursion level" is atomically
+            incremented to allow this.
```

Comment:
```suggestion
            return immediately. This is the difference between :class:`Lock` and
            :class:`!RLock`; :class:`Lock` handles this case the same as the previous,
            blocking until the lock can be acquired.
```

Be much more explicit and clear as to how these two differ. Also, remove the last sentence, which is an internal implementation detail.